File: cucumber-expressions/go/group.go
package cucumberexpressions

type Group struct {
	value    string
	start    int
	end      int
	children []*Group
}

func NewGroup(value string, start, end int, children []*Group) *Group {
	return &Group{
		value:    value,
		start:    start,
		end:      end,
		children: children,
	}
}

func (g *Group) Value() string {
	return g.value
}

func (g *Group) Start() int {
	return g.start
}

func (g *Group) End() int {
	return g.end
}

func (g *Group) Children() []*Group {
	return g.children
}

func (g *Group) Values() []string {
	if len(g.children) == 0 {
		return []string{g.value}
	}
	result := make([]string, len(g.children))
	for i, child := range g.children {
	  result[i] = child.Value()
	}
	return result
}

File: cucumber-expressions/javascript/src/group.ts
export default class Group {
  constructor(
    public readonly value: string | null,
    public readonly start: number,
    public readonly end: number,
    public readonly children: Group[]
  ) {}

  get values(): string[] {
    return (this.children.length === 0 ? [this] : this.children)
      .map(g => g.value)
      .filter(v => v !== null);
  }
}
